tools/nix/parts/tooling.nix

```
@@ -24,5 +24,13 @@ _: {
     formatter = pkgs.alejandra;
 
     packages.bats = pkgs.bats.withLibraries (p: [p.bats-assert p.bats-file p.bats-support]);
+
+    packages.shfmt-all = pkgs.writeShellApplication {
+      name = "shfmt-all";
+      runtimeInputs = with pkgs; [shfmt];
+      text = ''
+        shfmt -f <<<"$(git ls-files ':!:./demos/**' '*\.sh' '*\.bash' '*\.bats')" | xargs shfmt --diff
+      '';
```

If you call a binary in the body of the script, it needs to be in `runtimeInputs` as that's the entire point of this Nix construct. So we need to include Git and xargs, which is part of `findutils`. 

If you're not clear on why I suggested something in a previous conversation or code snippet, please **ask for clarification/my motive** rather than discarding the suggestion without any dialogue between us.

```suggestion
      runtimeInputs = with pkgs; [findutils gitMinimal shfmt];
      text = ''
        git ls-files ':!:./demos/**' '*\.sh' '*\.bash' '*\.bats' | xargs shfmt --diff
      '';
```

The proposed change to `text` seems to have exactly the same outcome with regards to output and exit status - was there something productive that `shfmt -f` was meant to be doing here? The help text merely says:

> -f, --find   recursively find all shell files and print the paths

And the GitHub thread you cited is from 2020, so probably the tool's behavior has evolved since then.

---

Yes thanks, adding `':!:./demos/**' '*\.sh' '*\.bash'` flag to git ls-files pretty much does the same thing as `shfmt -f`

---

I have pushed the recommendation